gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/CategoryServiceImpl.java

```diff
@@ -107,7 +107,7 @@ public List<CategoryEntity> findByPage(String page) {
     public CategoryEntity findById(final String id, final String environmentId) {
         try {
             LOGGER.debug("Find category by id : {}", id);
-            Optional<Category> category = categoryRepository.findById(id);
+            Optional<Category> category = categoryRepository.findById(id).filter(c -> c.getEnvironmentId().equalsIgnoreCase(environmentId));
```

Comment:
.orElseGet(categoryRepsitory.findByKey())

Reply:
I see but it adds complexity in the code for 2 reasons:
 - `categoryRepsitory.findByKey()` returns also an Optional
 - `categoryRepsitory.findByKey()` throws TechnicalManagementExceptions to handle
 
 So I prefer letting the code like this.